src/index.js

```diff
@@ -79,168 +75,166 @@ const idyll = (inputPath, opts, cb) => {
+              acc[name] = path.join(path.relative(TMP_PATH, compPath.dir), compPath.base).replace(/\\/g, '/');
+            } else if (customComponents.indexOf(name + '.js') > -1) {
+              acc[name] = path.relative(TMP_PATH, path.join(CUSTOM_COMPONENTS_FOLDER, name)).replace(/\\/g, '/');
+            } else if (components.indexOf(name + '.js') > -1) {
+              acc[name] = path.relative(TMP_PATH, path.join(DEFAULT_COMPONENTS_FOLDER, name)).replace(/\\/g, '/');
             } else {
-              data = require(path.join(DATA_FOLDER, source));
+              try {
+                // npm modules are required via relative paths to support working with a locally linked idyll
+                const compPath = path.parse(resolve.sync(name, {basedir: inputDirectory}));
+                acc[name] = path.join(path.relative(TMP_PATH, compPath.dir), compPath.base).replace(/\\/g, '/');
+              } catch (err) {
+                if (node[0].toLowerCase() !== node[0]) throw new Error(`Component named ${node[0]} could not be found.`)
+              }
             }
-            outputData[key] = data;
-            break;
-          case 'meta':
-            templateContext = {}
-            props.forEach((p) => {
-              templateContext[p[0]] = p[1][1];
-            })
-            break;
-        }
-      }
-      children.map(handleNode);
-    }
-    ast.map(handleNode);
+          }
 
-    fs.writeFileSync(COMPONENT_FILE, `module.exports = {\n${outputComponents.join(',\n')}\n} `);
-    fs.writeFileSync(DATA_FILE, `module.exports = ${JSON.stringify(outputData)}`);
-  }
+          return acc;
+        },
+        {}
+      )
+    }
 
+    return {
+      ast,
+      components: getComponents(ast),
+      data: getData(ast),
+      meta: getMeta(ast)
+    }
+  };
 
-  const build = (cb) => {
+  const build = (callback) => {
     process.env['NODE_ENV'] = 'production';
-    handleHTML();
-    // this path stuff is pretty ugly but necessary until client files don't rely on env vars
-    const clientDir = path.join(__dirname, 'client');
-    const visitorsDir = path.join(clientDir, 'visitors');
-    var b = browserify(path.join(clientDir, 'build.js'), {
+    var b = browserify(path.join(__dirname, 'client', 'build.js'), {
       transform: [
         [ babelify, { presets: [ reactPreset, es2015Preset ] } ],
-        [ envify, {
-          AST_FILE: path.join(path.relative(clientDir, TMP_PATH), path.parse(AST_FILE).base),
-          COMPONENT_FILE: path.join(path.relative(visitorsDir, TMP_PATH), path.parse(COMPONENT_FILE).base),
-          DATA_FILE: path.join(path.relative(visitorsDir, TMP_PATH), path.parse(DATA_FILE).base),
-          IDYLL_PATH
-        } ],
         [ brfs ]
       ]
     });
+    b.require(AST_FILE, {expose: '__IDYLL_AST__'});
+    b.require(COMPONENT_FILE, {expose: '__IDYLL_COMPONENTS__'});
+    b.require(DATA_FILE, {expose: '__IDYLL_DATA__'});
```

Comment:
Instead of relying on env vars, Browserify allows us to basically [alias these dependencies](https://github.com/substack/node-browserify#brequirefile-opts). This avoids some of the path craziness we had, removes the dependency on `envify`, and of course removes one more thing tying us to Node.